todo_app/templates/index.html

```diff
@@ -1,274 +1,170 @@
-      href="#nav-done" role="tab" aria-controls="nav-done" aria-selected="false">Done</a>
-    <a class="nav-item nav-link{{view_model.get_active_tab_class('All')}}" id="nav-all-tab" data-toggle="tab"
-      href="#nav-all" role="tab" aria-controls="nav-all" aria-selected="false">All</a>
-  </div>
-</nav>
+    <br></br>
+    <input class="form-control" type="text" id="myInput" onkeyup="searchTask()" placeholder="Search for task.."></input>
+    <br></br>
 
-<div class="tab-content" id="nav-tabContent">
-  <div class="tab-pane fade{{view_model.get_active_tab_content_class('To Do')}}" id="nav-todo" role="tabpanel"
-    aria-labelledby="nav-todo-tab">
-    <div class="table-responsive">
-      <table class="table table-bordered table-hover">
+    <details open>
+      <summary>TO DO</summary>
+      <table id="dtToDo" class="table-bordered table-hover table-striped table-sm" cellspacing="0" width="100%">
+        <caption></caption>
+        <thead>
           <tr>
-            <th scope="col"><a href="/sortby/title"
-                class="list-group-item list-group-item-action{{' active' if view_model.sort_order.column == 'title' else ''}}">Title</a>
-            </th>
-            <th scope="col"><a href="/sortby/status"
-                class="list-group-item list-group-item-action{{' active' if view_model.sort_order.column == 'status' else ''}}">Status</a>
-            </th>
-            <th scope="col"></th>
+            <th class="th-sm" style="display:none">#</th>
+            <th class="th-sm" style="width: 65%">Title</th>
+            <th class="th-sm" style="width: 10%">Due Date</th>
+            <th class="th-sm" style="width: 10%">Status</th>
+            <th class="th-sm" style="width: 15%">Actions</th>
           </tr>
+        </thead>
         <tbody>
-          {% for item in view_model.items_to_do%}
-          <tr>
-            <th scope="row">{{item.title}}</th>
-            <td>{{item.status}}</td>
+          {% for task in view_model.todo %}
+          <tr class="to-do-task">
+            <td style="display:none">{{ task.id }}</td>
+            <td>{{ task.name }}</td>
+            <td>{{ (task.due).strftime("%Y-%m-%d") }}</td>
+            <td>{{ task.status }}</td>
             <td>
-              {% if item.status == 'To Do'%}
-              <a href="{{url_for('move_item', action='doing', id=item.id)}}"
-                class="btn btn-outline-secondary btn-sm">Start</a>
-              {% elif item.status == 'Doing'%}
-              <a href="{{url_for('move_item', action='done', id=item.id)}}"
-                class="btn btn-outline-secondary btn-sm">Done</a>
-              {% elif item.status == 'Done'%}
-              <a href="{{url_for('move_item', action='doing', id=item.id)}}"
-                class="btn btn-outline-secondary btn-sm">Restart</a>
-              {% endif %}
+              <a class="btn btn-outline-success btn-group-sm" href='/edit/{{task.id}}'>Edit</button>
```

Comment:
You secured the server but it would be nicer to hide or disable these buttons when the user has insufficient permissions.